### ISSUE TITLE
'GENESIS' appears redundant

### ISSUE THREAD

**Issue by libscott** [bigchaindb/bigchaindb#683]
In [transaction.py](/bigchaindb/bigchaindb-common/blob/master/bigchaindb_common/transaction.py), there's no behavior associated with a GENESIS transaction. There is, however, [a bug](https://github.com/bigchaindb/bigchaindb-common/blob/master/bigchaindb_common/transaction.py#L683): it's trying to serialize an asset, which is not provided in the case of a genesis transaction, since the only purpose of the genesis transaction is to fulfill the condition that there are no empty blocks, so that the first `Vote` can have a `prev_block_id`.

[`is_genesis_block`](https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/util.py#L143) is the only case in bigchaindb[-common] where GENESIS is tested for.

Therefore it seems like it would be better to drop the `GENESIS` transaction altogether... And drop the genesis block, too.

From what I can see, the main _additional complexity_ associated with not having a genesis block is in [bigchain/core.py](https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/core.py#L636), to handle the case that there is no previous block. After that, you just get a `null` written in a Vote object when there is no previous block, and to find the first block you look for a vote with no `previous_block_id`.

In summary:

* Drop `GENESIS` from Transaction, now just `CREATE` and `TRANSFER`. Reduce ambiguity / complexity.
* Still have the rule that no blocks are empty

If this is not an option, let's say that someone wants to provide a special certificate in the genesis block of a public blockchain, it could be better to special case the genesis block so that it can be empty. But, equally it could be argued that the first CREATE transaction would be an appropriate place to store this.

**Comment by libscott** [comment]
Closing, relevant discussion in in #444
